### PR TITLE
Fixed leaks, certificate comparison and channel context cleanup

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

**Comment by ilammy** [comment]
I looked through this, seems okay to me.

File: libfreerdp/core/nla.c [hunk 1]
```diff
@@ -1215,7 +1217,7 @@ static BOOL nla_read_ts_credentials(rdpNla* nla, PSecBuffer ts_credentials)
 {
 	wStream* s;
 	int length;
-	int ts_password_creds_length;
+	int ts_password_creds_length = 0;
 	BOOL ret;
```
Comment (hardening) [inline libfreerdp/core/nla.c]: IMHO not need as this field is not used (and probably should be...)

File: libfreerdp/crypto/certificate.c [hunk 2]
```diff
@@ -184,7 +184,9 @@ static int certificate_data_match_legacy(rdpCertificateStore* certificate_store,
 					hostname, pline);
 			else if (strcmp(hostname, certificate_data->hostname) == 0)
 			{
-				match = strcmp(pline, certificate_data->fingerprint);
+				const int diff = strcmp(pline, certificate_data->fingerprint);
```
Comment (hardening) [inline libfreerdp/crypto/certificate.c]: const not needed

**Comment by akallabeth** [comment]
@hardening @bmiklautz anyone interested in merging this?

**Comment by hardening** [comment]
@akallabeth sorry to bother but could you remove the ```const int``` ?

**Comment by akallabeth** [comment]
@hardening Sorry, don't get it. What is the problem with a const variable?

**Comment by hardening** [comment]
@akallabeth well it's just that I don't see any utility for that, and that it's not the style of FreeRDP code...

**Comment by akallabeth** [comment]
@hardening ```const``` ist not style? The variable is constant, so there is no reason to hide that fact from the compiler...

**Comment by hardening** [comment]
well we've not used any const on scalar type. Ok merging it.

**Comment by akallabeth** [comment]
@hardening thx.

**Pull request merged**